src/shogun/mathematics/Random.cpp

```diff
@@ -120,7 +121,36 @@ void CRandom::init()
 	m_sfmt_32 = SG_MALLOC(sfmt_t, 1);
 	m_sfmt_64 = SG_MALLOC(sfmt_t, 1);
 	m_dsfmt = SG_MALLOC(dsfmt_t, 1);
-	reinit(m_seed);
+	reinit();
+
```

Comment:
I wonder: do we actually need the object as parameter in all this? If Random is a mixin, it should have access, no?

Reply:
yes